core/dbt/parser/models.py

```diff
@@ -26,46 +27,106 @@ def render_update(
     ) -> None:
         self.manifest._parsing_info.static_analysis_path_count += 1
 
-        # normal dbt run
-        if not flags.USE_EXPERIMENTAL_PARSER:
-            super().render_update(node, config)
+        # True roughly 1/100 times this function is called
+        sample: bool = random.randint(1, 101) == 100
 
-        # if the --use-experimental-parser flag was set
-        else:
+        # run the experimental parser if the flag is on or if we're sampling
+        if flags.USE_EXPERIMENTAL_PARSER or sample:
             try:
-                # run dbt jinja extractor (powered by tree-sitter + rust)
-                # throws an exception if it can't parse the source
-                res = py_extract_from_source(node.raw_sql)
+                experimentally_parsed = py_extract_from_source(node.raw_sql)
 
-                # since it doesn't need python jinja, fit the refs, sources, and configs
-                # into the node. Down the line the rest of the node will be updated with
-                # this information. (e.g. depends_on etc.)
+                # second config format
                 config_calls = []
-                for c in res['configs']:
+                for c in experimentally_parsed['configs']:
                     config_calls.append({c[0]: c[1]})
 
-                config._config_calls = config_calls
+                # format sources TODO change extractor to match this type
+                source_calls = []
+                for s in experimentally_parsed['sources']:
+                    source_calls.append([s[0], s[1]])
+                experimentally_parsed['sources'] = source_calls
+
+            except ExtractionError as e:
+                experimentally_parsed = e
+
+        # normal dbt run
+        if not flags.USE_EXPERIMENTAL_PARSER:
+            # normal rendering
+            super().render_update(node, config)
+            # if we're sampling, compare for correctness
+            if sample:
+                value = []
+                # experimental parser couldn't parse
+                if isinstance(experimentally_parsed, Exception):
+                    value += ["01_experimental_parser_couldn't_parse"]
```

Comment:
Nit: that apostrophe is going to be annoying
```suggestion
                    value += ["01_experimental_parser_cannot_parse"]
```

Reply:
very good point